src/components/views/EditPayment.jsx

```diff
@@ -0,0 +1,475 @@
+        });
+    }
+  }, [currentUser.id]);
+
+  // Update item of this item in milestone token
+  const updateAmount = () => {
+    const { token, selectedFiatType, date, fiatAmount } = payment;
+    if (!token.symbol || !selectedFiatType || payment.notCapped) return;
+
+    setLoadingAmount(true);
+    if (timer.current) {
+      clearTimeout(timer.current);
+    }
+
+    timer.current = setTimeout(async () => {
+      try {
+        setLoadingRate(true);
+        const res = await getConversionRates(date, token.symbol, selectedFiatType);
+        const rate = res.rates[selectedFiatType];
+        if (rate && isMounted.current) {
+          conversionRateTimestamp.current = res.timestamp;
+          setMaxAmount(new BigNumber(fiatAmount).div(rate));
+        } else {
+          throw new Error('Rate not found');
+        }
+      } catch (e) {
+        const message = `Sadly we were unable to get the exchange rate. Please try again after refresh.`;
+
+        ErrorHandler(e, message);
+        setMaxAmount(0);
+      } finally {
+        setLoadingRate(false);
+        setLoadingAmount(false);
+      }
+    }, WAIT_INTERVAL);
+  };
+
+  useEffect(() => {
+    updateAmount();
+  }, [payment.token, payment.fiatAmount, payment.date, payment.selectedFiatType]);
+
+  const handleInputChange = event => {
+    const { name, value, type, checked } = event.target;
+    if (type === 'checkbox') {
+      setPayment({ ...payment, [name]: checked });
+    } else {
+      setPayment({ ...payment, [name]: value });
+    }
+  };
+
+  const handleSelectCurrency = (_, option) => {
+    handleInputChange({
+      target: { name: 'selectedFiatType', value: option.value },
+    });
+  };
+
+  const handleSelectToken = token => {
+    handleInputChange({
+      target: { name: 'token', value: token },
+    });
+  };
+
+  const handleDatePicker = dateString => {
+    handleInputChange({ target: { name: 'date', value: dateString } });
+  };
+
+  const setPicture = address => {
+    handleInputChange({ target: { name: 'image', value: address } });
+  };
+
+  const submit = async () => {
+    const authenticated = await authenticateUser(currentUser, false);
+
+    if (authenticated) {
+      if (userIsCampaignOwner && !isForeignNetwork) {
+        displayForeignNetRequiredWarning();
+        return;
+      }
+
+      const {
+        title,
+        description,
+        image,
+        recipientAddress,
+        notCapped,
+        fiatAmount,
+        selectedFiatType,
+        token,
+        date,
+        donateToDac,
+      } = payment;
+
+      milestone.parentProjectId = campaign.projectId;
+      milestone.title = title;
+      milestone.description = description;
+      milestone.recipientAddress = recipientAddress;
+      milestone.image = image;
+      milestone.token = token;
+      milestone.dacId = donateToDac ? config.defaultDacId : 0;
+      // TODO: We should have ability to delete fiatAmount for uncapped milestones
+
+      if (!notCapped) {
```

Comment:
We should have an else statement for this condition and clear max amount and other fields if milestone is not capped.

@RamRamez refer to this
https://github.com/Giveth/giveth-dapp/pull/2126#pullrequestreview-660389475